Add linux-specific `core.sys.linux.file` module

Burgos opened the pull request:
This adds core.sys.linux.file module which contains
flock function definition and values of the parameters
to pass.

CyberShadow commented:
I believe this should be `core.sys.linux.sys.file` to follow the naming scheme for `sys/file.h`.

CyberShadow commented:
BTW, unless you specifically need non-advisory locks, we already have definitions for fcntl locks, also exposed via `std.stdio.File.lock()` etc.

CyberShadow commented:
Makefiles and file lists need to be modified when adding new modules. See all occurrences of an existing module name for an example, e.g. `git grep timerfd | grep -v ^src/`.

Burgos commented:
> I believe this should be core.sys.linux.sys.file to follow the naming scheme for sys/file.h.

Thanks, I got fooled by the `sys` prefix.

Updated with moved location and updated `make/COPY`, `win32.mak` and `win64.mak` files.

CyberShadow commented on src/core/sys/linux/sys/file.d:
```diff
@@ -0,0 +1,35 @@
+/// Unlock
+enum LOCK_UN              = 0x08;
+/// Atomic update
+enum __LOCK_ATOMIC        = 0x16;
```
This one isn't in my `file.h`. Is it Debian-specific?

CyberShadow replied:
I think this one is only implemented in the GNU Hurd kernel.

Burgos replied:
Hmm, I got it from mine (Ubuntu's) file.h, but I don't see it on glibc's upstream file.h, so I'll remove it.

Thanks and good catch!

Burgos commented:
Updated with `__LOCK_ATOMIC` removed.

DmitryOlshansky commented:
Auto-merge toggled on

DmitryOlshansky commented:
Looks good apart from horrible windows makefiles... but that's separate problem.

andralex commented:
thx!
